### ISSUE TITLE
StrictMode warnings

### ISSUE THREAD

**Issue by Hypnosphi** [reactjs/react-lifecycles-compat#23]
If a polyfilled component uses `getSnapshotBeforeUpdate` but doesn't use `static getDerivedStateFromProps`, it produces a warning in strict mode:

https://codesandbox.io/s/v6rv7v59l3
```
Warning: Unsafe lifecycle methods were found within a strict-mode tree:
    in App


componentWillUpdate: Please update the following components to use componentDidUpdate instead: MyComponent
```

The reason is that React only checks for suppressed warning in `cWM`, which isn't applied for `gSBU`
https://github.com/facebook/react/blob/8ec0e4a99df76c0ff1779cac4f2eaaaf35a6b5bb/packages/react-reconciler/src/ReactStrictModeWarnings.js#L244

**Comment by bvaughn** [comment]
Ah, good catch. I believe that check was added before the decision to polyfill gSBU was made.

**Comment by bvaughn** [comment]
This has been fixed via facebook/react/pull/12647 and will go out with the next React bugfix release. Thanks for pointing it out!
